### ISSUE TITLE
Add app to the main F-droid repo

### ISSUE THREAD

**Issue by niporasomo** [thunder-app/thunder#347]
**Is your feature request related to a problem? Please describe.**

It would be awesome if the app could be added to the main F-droid repo! More people could have a chance to see it too if they don't have the IzzyOnDroid repo added. I'm not sure if it would be possible to do with Thunder, but I think the person who maintains IzzyOnDroid is also a maintainer with F-droid and might be able to help?

Thanks for your time and for making an awesome app!

**Comment by hjiangsu** [comment]
I'll have to see what the process is to get it on F-Droid!

Thankfully, IzzyOnDroid has helped a lot with getting Thunder on there, but I'm not too familiar with how F-Droid works. On the other hand, I've been working on getting Thunder on the Play Store, so that is something that's happening in the background to help bring more people here

**Comment by niporasomo** [comment]
Awesome! I think this link has their criteria to be accepted, and I'm sure IzzyOnDroid can lend a hand too:

https://f-droid.org/docs/Inclusion_Policy/

Also love all the development work going on with the app, especially since so many people have been moving over to Lemmy recently. Google Play, Izzy and F-droid should definitely get more attention for Thunder.

**Comment by RustoMCSpit** [comment]
is revolt foss?

**Comment by shuvashish76** [comment]
RFP: https://gitlab.com/fdroid/rfp/-/issues/2502

**Comment by IzzySoft** [comment]
Fastlane is not found by he F-Droid process as it is in the wrong place: `/android/fastlane/metadata/android`, it must be moved one level up. Can you do that? (and please let me know if it was done, so I adjust it for my repo; F-Droid looks in pre-defined "standard locations", with my repo this is configurable).

**Comment by hjiangsu** [comment]
Just going to circle back to this: I haven't done testing yet on whether or not moving the `fastlane` folder up a level will affect anything. If this is a blocker for getting it approved on F-Droid, then maybe we can do a test and see if its viable!

I think it should be okay since it looks like Liftoff also has the fastlane directory on the top level.

**Comment by IzzySoft** [comment]
Yes, please do the test. F-Droid only finds those details at the defined locations (see my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) for details, which outlines the 3 possible locations). The most common one is what I mentioned above, and the majority of apps use it that way.

**Comment by hjiangsu** [comment]
Sorry for the long delay on this, but I've moved the `fastlane` directory up to the root directory. I hope this resolves the issue!

@IzzySoft Are you able to double check if this is valid without the need to generate a new build (i.e. clone from the `develop` branch)? If not, then we can wait until the next pre-release/general release to check this.

**Comment by IzzySoft** [comment]
It looks fine to me. I've adjusted that in my setup to pull from the develop branch and the new fastlane location; remember to let me know when I should change that as with my updater the location is configurable.

As for the "main F-Droid repo": I'm currently [not active at F-Droid](https://gitlab.com/fdroid/admin/-/issues/447). But I doubt anything concerning that has changed in the meantime, so they'll pick up fastlane from the same commit they build the APK from.

**Comment by hjiangsu** [comment]
> It looks fine to me. I've adjusted that in my setup to pull from the develop branch and the new fastlane location; remember to let me know when I should change that as with my updater the location is configurable.

Thanks again for checking! And I'm sorry to hear that. Thanks for all the work you've done and best wishes for your future endeavours 😊

I'll keep this issue open until everything is confirmed to be working.

**Comment by IzzySoft** [comment]
> Thanks for all the work you've done and best wishes for your future endeavours

Gladly, and thanks – I'll concentrate on my repo currently, so expect "things to come" there :wink:

**Comment by Thewisem** [comment]
Is there any development on this

**Comment by hjiangsu** [comment]
Hey, I just added a comment to the relevant [upstream issue for F-Droid](https://gitlab.com/fdroid/rfp/-/issues/2502). Aside from this, there's not much else I can do!

**Comment by shuvashish76** [comment]
Someone needs to create a merge request. Probably F-Droid maintainers are busy with other things... @hjiangsu why not you try it yourself? Here are the [instructions](https://gitlab.com/fdroid/fdroiddata/blob/master/CONTRIBUTING.md#merge-requests).

**Comment by hjiangsu** [comment]
Ahh I see - I can look into that. Thanks!

**Comment by hjiangsu** [comment]
For devs:

So a bit of an update, I'm working on trying to get a merge request for F-Droid, but am running into an issue where F-Droid is detecting an invalid `maven` URL for `background_fetch`. This is most likely because of the way we set up the dependency for `background_fetch` in our `build.gradle`. I'm assuming it fails because F-Droid cannot confirm the contents of it?

```
maven {
    url "${project(':background_fetch').projectDir}/libs"
}
```

For reference:
- This is my fork: https://gitlab.com/hjiangsu/fdroiddata/-/tree/com.hjiangsu.thunder?ref_type=heads
- This is the build error: https://gitlab.com/hjiangsu/fdroiddata/-/jobs/6995781519
- The `background_fetch` repo: https://github.com/transistorsoft/flutter_background_fetch

@micahmo I'm not sure if there's a way to get around this issue without removing `background_fetch` entirely from the F-Droid build (which would require us to maintain a separate branch which is not ideal). Do you have any thoughts on this?

**Comment by shuvashish76** [comment]
@linsui ping for the help

**Comment by linsui** [comment]
See https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/dev.imranr.obtainium.fdroid.yml.

**Comment by hjiangsu** [comment]
Thanks, I'll check that out!

**Comment by Coehill** [comment]
Looking forward to this being on F-Droid! There is a big overlap of Lemmy users and F-Droid users since both appeal to the FOSS community.

**Comment by linsui** [comment]
background_fetch is published to maven central now so its own maven repo is not used anymore.
